tools/test

```diff
@@ -185,7 +185,7 @@ run_lint() {
     local files
     files=( $(apply_eslintignore "$@") )
     (( ${#files[@]} )) || return 0
-    eslint ${fix:+--fix} --max-warnings=0 "${files[@]}"
+    eslint ${fix:+--fix} --report-unused-disable-directives --max-warnings=0 "${files[@]}"
```

Comment:
Hmm interesting; what's the reasoning for putting this here instead of in the eslintrc?

I think one difference in resulting behavior will be that the reports won't appear in the IDE, only when running `tools/test`.

Reply:
I almost wrote down the reason, but it was getting complicated…if you want, you can try it, and be sure to run both `tools/test prettier` and `tools/fmt`. I don't think I found a way to put it in the config so that it'd work.

ISTR if we put it in the config, `prettier-eslint` ends up passing it to an `eslint` API (a constructor function?) in the wrong form. When I got error output suggesting that, I tried writing down the expected form in the config file. But then the config file didn't pass validation.

Reply:
Ha, I see. If I write `reportUnusedDisableDirectives: true`, then ESLint is happy (and I get warnings as expected in the IDE), but:
```
$ tools/test prettier --diff @
Running prettier...
prettier-eslint [ERROR]: There was trouble creating the ESLint CLIEngine.
prettier-eslint-cli [ERROR]: There was an error formatting "/home/greg/z/mobile/src/events/doEventActionSideEffects.js": 
    Error: Invalid Options:
    - 'reportUnusedDisableDirectives' must be any of "error", "warn", "off", and null.
…
```

If I follow that error message and write, say, `reportUnusedDisableDirectives: error`, then `tools/test prettier` is happy, but:
```
$ tools/test lint --diff @
Running lint...

Oops! Something went wrong! :(

ESLint: 8.16.0

Error: ESLint configuration in .eslintrc.yaml is invalid:
	- Property "reportUnusedDisableDirectives" is the wrong type (expected boolean but got `"error"`).
…
```

Odd because looking at the stack traces, it seems like it's two different APIs of ESLint that are requiring incompatible ways of writing that option:
```
prettier-eslint-cli [ERROR]: There was an error formatting "/home/greg/z/mobile/src/events/doEventActionSideEffects.js": 
    Error: Invalid Options:
    - 'reportUnusedDisableDirectives' must be any of "error", "warn", "off", and null.
        at processOptions (/home/greg/z/mobile/node_modules/eslint/lib/eslint/eslint.js:281:15)
        at new ESLint (/home/greg/z/mobile/node_modules/eslint/lib/eslint/eslint.js:428:34)
        at getESLint (/home/greg/z/mobile/node_modules/prettier-eslint/dist/utils.js:386:12)
        …
```
vs.
```
ESLint: 8.16.0

Error: ESLint configuration in .eslintrc.yaml is invalid:
	- Property "reportUnusedDisableDirectives" is the wrong type (expected boolean but got `"error"`).

    at ConfigValidator.validateConfigSchema (/home/greg/z/mobile/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2156:19)
    at ConfigArrayFactory._normalizeConfigData (/home/greg/z/mobile/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2982:19)
    at ConfigArrayFactory.loadInDirectory (/home/greg/z/mobile/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2850:33)
    …
    at CascadingConfigArrayFactory.getConfigArrayForFile (/home/greg/z/mobile/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3754:18)
    at FileEnumerator._iterateFilesWithFile (/home/greg/z/mobile/node_modules/eslint/lib/cli-engine/file-enumerator.js:366:43)
```

:shrug: Not worth spending the time to debug, I think -- we'll just tolerate not getting this particular warning in the IDE.